### ISSUE TITLE
Unable to utilize the az ml workspace update command

### ISSUE THREAD

**Issue by mstetka-fr** [Azure/azure-cli-extensions#7699]
### Describe the bug

Using the latest version of the azure-cli and ml extension, the az ml workspace update command does not work in any circumstance, even if the command does not issue any changes:

```
C:\az ml workspace update -n synapse-dev-cn-mlworkspace -g synapse-dev
The command failed with an unexpected error. Here is the traceback:
name 'select_load_function' is not defined
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 664, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 731, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 701, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 334, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 240, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 453, in cached_put
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 447, in _put_operation
  File "C:\Users\mstetka\.azure\cliextensions\ml\azext_mlv2\manual\custom\workspace.py", line 212, in ml_workspace_update
    workspace = select_load_function()(file, params_override=params_override)
                ^^^^^^^^^^^^^^^^^^^^
NameError: name 'select_load_function' is not defined
```

### Related command

`az ml workspace update -n "example_workspace_name" -g "example_resource_group_name"`

### Errors

The command failed with an unexpected error. Here is the traceback:
name 'select_load_function' is not defined
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 664, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 731, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 701, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 334, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 240, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 453, in cached_put
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 447, in _put_operation
  File "C:\Users\mstetka\.azure\cliextensions\ml\azext_mlv2\manual\custom\workspace.py", line 212, in ml_workspace_update
    workspace = select_load_function()(file, params_override=params_override)
                ^^^^^^^^^^^^^^^^^^^^
NameError: name 'select_load_function' is not defined

### Issue script & Debug output

DEBUG: cli.knack.cli: Command arguments: ['ml', 'workspace', 'update', '-n', 'synapse-dev-cn-mlworkspace', '-g', 'synapse-dev', '--debug']
DEBUG: cli.knack.cli: __init__ debug log:
Cannot enable color.
DEBUG: cli.knack.cli: Event: Cli.PreExecute []
DEBUG: cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x0000028838CBB880>, <function OutputProducer.on_global_arguments at 0x0000028838E460C0>, <function CLIQuery.on_global_arguments at 0x0000028838E73C40>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
DEBUG: cli.azure.cli.core: Modules found from index for 'ml': ['azext_mlv2']
DEBUG: cli.azure.cli.core: Loading command modules:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands
DEBUG: cli.azure.cli.core: Total (0)                 0.000         0         0
DEBUG: cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
DEBUG: cli.azure.cli.core: Loading extensions:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
DEBUG: azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the default cloud configuration: 'AzureCloud'.
DEBUG: azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the default cloud configuration: 'AzureCloud'.
DEBUG: azext_mlv2.manual.vendored_curated_sdk.azure.ai.ml._azure_environments: Using the default cloud configuration: 'AzureCloud'.
DEBUG: cli.azure.cli.core: ml                        2.112        24       160  C:\Users\mstetka\.azure\cliextensions\ml
DEBUG: cli.azure.cli.core: Total (1)                 2.112        24       160  
DEBUG: cli.azure.cli.core: Loaded 24 groups, 160 commands.
DEBUG: cli.azure.cli.core: Found a match in the command table.
DEBUG: cli.azure.cli.core: Raw command  : ml workspace update
DEBUG: cli.azure.cli.core: Command table: ml workspace update
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x000002883BD9E340>]
DEBUG: cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to 'C:\Users\mstetka\.azure\commands\2024-06-10.11-46-37.ml_workspace_update.15612.log'.
INFO: az_command_data_logger: command args: ml workspace update -n {} -g {} --debug
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x000002883BDD67A0>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x000002883BE087C0>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x000002883BE08900>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x0000028838E46160>, <function CLIQuery.handle_query_parameter at 0x0000028838E73CE0>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x000002883BE08860>]
INFO: az_command_data_logger: extension name: ml
INFO: az_command_data_logger: extension version: 2.26.1
DEBUG: cli.azure.cli.core.commands.client_factory: Getting management service client client_type=AzureMachineLearningWorkspaces
DEBUG: cli.azure.cli.core.auth.persistence: build_persistence: location='C:\\Users\\mstetka\\.azure\\msal_token_cache.bin', encrypt=True
DEBUG: cli.azure.cli.core.auth.binary_cache: load: C:\Users\mstetka\.azure\msal_http_cache.bin
DEBUG: urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
INFO: msal.authority: Initializing with Entra authority: https://login.microsoftonline.com/<redacted_subscription_id>
DEBUG: msal.authority: openid_config("https://login.microsoftonline.com/<redacted_subscription_id>/v2.0/.well-known/openid-configuration") = {'token_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/token', 'token_endpoint_auth_methods_supported': ['client_secret_post', 'private_key_jwt', 'client_secret_basic'], 'jwks_uri': 'https://login.microsoftonline.com/<redacted_subscription_id>/discovery/v2.0/keys', 'response_modes_supported': ['query', 'fragment', 'form_post'], 'subject_types_supported': ['pairwise'], 'id_token_signing_alg_values_supported': ['RS256'], 'response_types_supported': ['code', 'id_token', 'code id_token', 'id_token token'], 'scopes_supported': ['openid', 'profile', 'email', 'offline_access'], 'issuer': 'https://login.microsoftonline.com/<redacted_subscription_id>/v2.0', 'request_uri_parameter_supported': False, 'userinfo_endpoint': 'https://graph.microsoft.com/oidc/userinfo', 'authorization_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/authorize', 'device_authorization_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/devicecode', 'http_logout_supported': True, 'frontchannel_logout_supported': True, 'end_session_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/logout', 'claims_supported': ['sub', 'iss', 'cloud_instance_name', 'cloud_instance_host_name', 'cloud_graph_host_name', 'msgraph_host', 'aud', 'exp', 'iat', 'auth_time', 'acr', 'nonce', 'preferred_username', 'name', 'tid', 'ver', 'at_hash', 'c_hash', 'email'], 'kerberos_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/kerberos', 'tenant_region_scope': 'NA', 'cloud_instance_name': 'microsoftonline.com', 'cloud_graph_host_name': 'graph.windows.net', 'msgraph_host': 'graph.microsoft.com', 'rbac_url': 'https://pas.windows.net'}
DEBUG: msal.application: Broker enabled? True
DEBUG: cli.azure.cli.core.commands.client_factory: Getting management service client client_type=_ml_client_cli
DEBUG: urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
INFO: msal.authority: Initializing with Entra authority: https://login.microsoftonline.com/<redacted_subscription_id>
DEBUG: msal.authority: openid_config("https://login.microsoftonline.com/<redacted_subscription_id>/v2.0/.well-known/openid-configuration") = {'token_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/token', 'token_endpoint_auth_methods_supported': ['client_secret_post', 'private_key_jwt', 'client_secret_basic'], 'jwks_uri': 'https://login.microsoftonline.com/<redacted_subscription_id>/discovery/v2.0/keys', 'response_modes_supported': ['query', 'fragment', 'form_post'], 'subject_types_supported': ['pairwise'], 'id_token_signing_alg_values_supported': ['RS256'], 'response_types_supported': ['code', 'id_token', 'code id_token', 'id_token token'], 'scopes_supported': ['openid', 'profile', 'email', 'offline_access'], 'issuer': 'https://login.microsoftonline.com/<redacted_subscription_id>/v2.0', 'request_uri_parameter_supported': False, 'userinfo_endpoint': 'https://graph.microsoft.com/oidc/userinfo', 'authorization_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/authorize', 'device_authorization_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/devicecode', 'http_logout_supported': True, 'frontchannel_logout_supported': True, 'end_session_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/logout', 'claims_supported': ['sub', 'iss', 'cloud_instance_name', 'cloud_instance_host_name', 'cloud_graph_host_name', 'msgraph_host', 'aud', 'exp', 'iat', 'auth_time', 'acr', 'nonce', 'preferred_username', 'name', 'tid', 'ver', 'at_hash', 'c_hash', 'email'], 'kerberos_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/kerberos', 'tenant_region_scope': 'NA', 'cloud_instance_name': 'microsoftonline.com', 'cloud_graph_host_name': 'graph.windows.net', 'msgraph_host': 'graph.microsoft.com', 'rbac_url': 'https://pas.windows.net'}
DEBUG: msal.application: Broker enabled? True
DEBUG: cli.azure.cli.core.auth.credential_adaptor: CredentialAdaptor.get_token: scopes=('https://management.azure.com/.default',), kwargs={}
DEBUG: cli.azure.cli.core.auth.msal_authentication: UserCredential.get_token: scopes=('https://management.azure.com/.default',), claims=None, kwargs={}
DEBUG: msal.application: Cache hit an AT
DEBUG: msal.telemetry: Generate or reuse correlation_id: 626a3126-1703-403e-9e44-e287ee509480
DEBUG: azure.core.pipeline.policies._universal: Request URL: 'https://management.azure.com/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.MachineLearningServices/workspaces/synapse-dev-cn-mlworkspace?api-version=2023-08-01-preview'
Request method: 'GET'
Request headers:
    'Accept': 'application/json'
    'x-ms-client-request-id': '9dd142ea-2740-11ef-b9cb-f66add85907b'
    'User-Agent': 'azureml-cli-v2/2.26.1 azure-ai-ml/1.16.1 azsdk-python-mgmt-machinelearningservices/0.1.0 Python/3.11.8 (Windows-10-10.0.19045-SP0)'
    'Authorization': 'Bearer *'
Request body:
This request has no body
DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): management.azure.com:443
DEBUG: urllib3.connectionpool: https://management.azure.com:443 "GET /subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.MachineLearningServices/workspaces/synapse-dev-cn-mlworkspace?api-version=2023-08-01-preview HTTP/1.1" 200 5762
DEBUG: azure.core.pipeline.policies._universal: Response status: '200'
Response headers:
    'Cache-Control': 'no-cache'
    'Pragma': 'no-cache'
    'Content-Length': '5762'
    'Content-Type': 'application/json; charset=utf-8'
    'Expires': '-1'
    'Vary': 'Accept-Encoding'
    'x-ms-ratelimit-remaining-subscription-global-reads': '3749'
    'x-ms-ratelimit-remaining-subscription-reads': '249'
    'Request-Context': 'appId=cid-v1:2d2e8e63-272e-4b3c-8598-4ee570a0e70d'
    'x-ms-response-type': 'standard'
    'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
    'X-Content-Type-Options': 'nosniff'
    'x-aml-cluster': 'vienna-centralus-01'
    'x-request-time': '0.031'
    'x-ms-request-id': '3ef6bcee-d787-48c7-b327-0c49a048000b'
    'x-ms-correlation-request-id': '3ef6bcee-d787-48c7-b327-0c49a048000b'
    'x-ms-routing-request-id': 'NORTHCENTRALUS:20240610T154638Z:3ef6bcee-d787-48c7-b327-0c49a048000b'
    'X-Cache': 'CONFIG_NOCACHE'
    'X-MSEdge-Ref': 'Ref A: B108384A717745F38752F4E0E9923368 Ref B: CH1AA2020620049 Ref C: 2024-06-10T15:46:37Z'
    'Date': 'Mon, 10 Jun 2024 15:46:37 GMT'
Response content:
{
  "id": "/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.MachineLearningServices/workspaces/synapse-dev-cn-mlworkspace",
  "name": "synapse-dev-cn-mlworkspace",
  "type": "Microsoft.MachineLearningServices/workspaces",
  "location": "centralus",
  "tags": {
    "application": "Synapse",
    "applicationType": "Analytics",
    "environment": "dev",
    "terraform": "synapse"
  },
  "etag": null,
  "properties": {
    "friendlyName": "",
    "description": "",
    "storageAccount": "/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.Storage/storageAccounts/syndcnmlwstg",
    "keyVault": "/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.Keyvault/vaults/syndcn-mlw-keyvault",
    "applicationInsights": "/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.insights/components/synapse-dev-cn-appinsights",
    "hbiWorkspace": false,
    "tenantId": "<redacted_subscription_id>",
    "imageBuildCompute": null,
    "provisioningState": "Succeeded",
    "managedNetwork": {
      "changeableIsolationModes": [
        "AllowInternetOutbound",
        "AllowOnlyApprovedOutbound"
      ],
      "isolationMode": "Disabled"
    },
    "v1LegacyMode": false,
    "softDeleteEnabled": false,
    "containerRegistry": "/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.ContainerRegistry/registries/c35ef67bfc3e4e3591a7b3f45bb6beee",
    "creationTime": "2024-02-07T20:18:26.7301501Z",
    "notebookInfo": {
      "resourceId": "<redacted>",
      "fqdn": "ml-synapse-dev-c-centralus-<redacted_workspace_id>.centralus.notebooks.azure.net",
      "isPrivateLinkEnabled": true,
      "notebookPreparationError": null
    },
    "storageHnsEnabled": false,
    "workspaceId": "<redacted_workspace_id>",
    "linkedModelInventoryArmId": null,
    "privateLinkCount": 2,
    "privateEndpointConnections": [
      {
        "id": "/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.MachineLearningServices/workspaces/synapse-dev-cn-mlworkspace/privateEndpointConnections/synapse-dev-cn-mlworkspace.60ae8f55-f803-40fc-980a-5a229af1f6ce",
        "location": "centralus",
        "properties": {
          "provisioningState": "Succeeded",
          "privateEndpoint": {
            "id": "/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.Network/privateEndpoints/synapse-dev-cn-mlworkspace-userdev-endpoint",
            "subnetArmId": "/subscriptions/<redacted_subscription_id>/resourceGroups/common-master/providers/Microsoft.Network/virtualNetworks/common-master-cn-vnet-01/subnets/common-master-cn-subnet-userdev-01"
          },
          "privateLinkServiceConnectionState": {
            "status": "Approved",
            "description": "Auto-Approved",
            "actionsRequired": "None"
          }
        }
      },
      {
        "id": "/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.MachineLearningServices/workspaces/synapse-dev-cn-mlworkspace/privateEndpointConnections/synapse-dev-cn-mlworkspace.4cd5f538-eff6-40fa-abd3-1125609ef86d",
        "location": "centralus",
        "properties": {
          "provisioningState": "Succeeded",
          "privateEndpoint": {
            "id": "/subscriptions/da768350-cf5e-4395-8555-1e83b718f151/resourceGroups/vnet-da768350-centralus-46-rg/providers/Microsoft.Network/privateEndpoints/synapse-dev-cn-synapse-workspace.Machine-Learning-Workspace-Managed-Endpoint",
            "subnetArmId": "/subscriptions/da768350-cf5e-4395-8555-1e83b718f151/resourceGroups/vnet-da768350-centralus-46-rg/providers/Microsoft.Network/virtualNetworks/vnet-da768350-centralus-46/subnets/PrivateEndpointsSubnet"
          },
          "privateLinkServiceConnectionState": {
            "status": "Approved",
            "description": "Manually Approved",
            "actionsRequired": "None"
          }
        }
      }
    ],
    "publicNetworkAccess": "Disabled",
    "ipAllowlist": [],
    "discoveryUrl": "https://<redacted_workspace_id>.workspace.centralus.api.azureml.ms/discovery/workspaces/<redacted_workspace_id>",
    "mlFlowTrackingUri": "azureml://<redacted_workspace_id>.workspace.centralus.api.azureml.ms/mlflow/v1.0/subscriptions/<redacted_subscription_id>/resourceGroups/synapse-dev/providers/Microsoft.MachineLearningServices/workspaces/synapse-dev-cn-mlworkspace",
    "serverlessComputeSettings": {
      "serverlessComputeCustomSubnet": "/subscriptions/<redacted_subscription_id>/resourceGroups/common-master/providers/Microsoft.Network/virtualNetworks/common-master-cn-vnet-01/subnets/common-master-cn-subnet-userdev-01",
      "serverlessComputeNoPublicIP": true
    },
    "EnableSoftwareBillOfMaterials": false,
    "sdkTelemetryAppInsightsKey": "8e30de65-9fdd-423f-86b2-93e5277cf4c0",
    "enableDataIsolation": false,
    "systemDatastoresAuthMode": "accesskey",
    "enableServiceSideCMKEncryption": false
  },
  "identity": {
    "type": "SystemAssigned",
    "principalId": "c2687fc3-c42c-4ce1-8651-9468fab55cd2",
    "tenantId": "<redacted_subscription_id>"
  },
  "kind": "Default",
  "sku": {
    "name": "Basic",
    "tier": "Basic"
  },
  "systemData": {
    "createdAt": "2024-02-07T20:18:26.7301501Z",
    "lastModifiedAt": "2024-06-07T19:39:47.8743241Z",
    "lastModifiedBy": "ADMINDP@fleetresponse.onmicrosoft.com",
    "lastModifiedByType": "User"
  }
}
DEBUG: cli.azure.cli.core.commands.client_factory: Getting management service client client_type=AzureMachineLearningWorkspaces
DEBUG: urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
INFO: msal.authority: Initializing with Entra authority: https://login.microsoftonline.com/<redacted_subscription_id>
DEBUG: msal.authority: openid_config("https://login.microsoftonline.com/<redacted_subscription_id>/v2.0/.well-known/openid-configuration") = {'token_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/token', 'token_endpoint_auth_methods_supported': ['client_secret_post', 'private_key_jwt', 'client_secret_basic'], 'jwks_uri': 'https://login.microsoftonline.com/<redacted_subscription_id>/discovery/v2.0/keys', 'response_modes_supported': ['query', 'fragment', 'form_post'], 'subject_types_supported': ['pairwise'], 'id_token_signing_alg_values_supported': ['RS256'], 'response_types_supported': ['code', 'id_token', 'code id_token', 'id_token token'], 'scopes_supported': ['openid', 'profile', 'email', 'offline_access'], 'issuer': 'https://login.microsoftonline.com/<redacted_subscription_id>/v2.0', 'request_uri_parameter_supported': False, 'userinfo_endpoint': 'https://graph.microsoft.com/oidc/userinfo', 'authorization_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/authorize', 'device_authorization_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/devicecode', 'http_logout_supported': True, 'frontchannel_logout_supported': True, 'end_session_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/logout', 'claims_supported': ['sub', 'iss', 'cloud_instance_name', 'cloud_instance_host_name', 'cloud_graph_host_name', 'msgraph_host', 'aud', 'exp', 'iat', 'auth_time', 'acr', 'nonce', 'preferred_username', 'name', 'tid', 'ver', 'at_hash', 'c_hash', 'email'], 'kerberos_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/kerberos', 'tenant_region_scope': 'NA', 'cloud_instance_name': 'microsoftonline.com', 'cloud_graph_host_name': 'graph.windows.net', 'msgraph_host': 'graph.microsoft.com', 'rbac_url': 'https://pas.windows.net'}
DEBUG: msal.application: Broker enabled? True
DEBUG: cli.azure.cli.core.commands.client_factory: Getting management service client client_type=_ml_client_cli
DEBUG: urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
INFO: msal.authority: Initializing with Entra authority: https://login.microsoftonline.com/<redacted_subscription_id>
DEBUG: msal.authority: openid_config("https://login.microsoftonline.com/<redacted_subscription_id>/v2.0/.well-known/openid-configuration") = {'token_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/token', 'token_endpoint_auth_methods_supported': ['client_secret_post', 'private_key_jwt', 'client_secret_basic'], 'jwks_uri': 'https://login.microsoftonline.com/<redacted_subscription_id>/discovery/v2.0/keys', 'response_modes_supported': ['query', 'fragment', 'form_post'], 'subject_types_supported': ['pairwise'], 'id_token_signing_alg_values_supported': ['RS256'], 'response_types_supported': ['code', 'id_token', 'code id_token', 'id_token token'], 'scopes_supported': ['openid', 'profile', 'email', 'offline_access'], 'issuer': 'https://login.microsoftonline.com/<redacted_subscription_id>/v2.0', 'request_uri_parameter_supported': False, 'userinfo_endpoint': 'https://graph.microsoft.com/oidc/userinfo', 'authorization_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/authorize', 'device_authorization_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/devicecode', 'http_logout_supported': True, 'frontchannel_logout_supported': True, 'end_session_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/oauth2/v2.0/logout', 'claims_supported': ['sub', 'iss', 'cloud_instance_name', 'cloud_instance_host_name', 'cloud_graph_host_name', 'msgraph_host', 'aud', 'exp', 'iat', 'auth_time', 'acr', 'nonce', 'preferred_username', 'name', 'tid', 'ver', 'at_hash', 'c_hash', 'email'], 'kerberos_endpoint': 'https://login.microsoftonline.com/<redacted_subscription_id>/kerberos', 'tenant_region_scope': 'NA', 'cloud_instance_name': 'microsoftonline.com', 'cloud_graph_host_name': 'graph.windows.net', 'msgraph_host': 'graph.microsoft.com', 'rbac_url': 'https://pas.windows.net'}
DEBUG: msal.application: Broker enabled? True
DEBUG: cli.azure.cli.core.azclierror: Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 664, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 731, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 701, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 334, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 240, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 453, in cached_put
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 447, in _put_operation
  File "C:\Users\mstetka\.azure\cliextensions\ml\azext_mlv2\manual\custom\workspace.py", line 212, in ml_workspace_update
    workspace = select_load_function()(file, params_override=params_override)
                ^^^^^^^^^^^^^^^^^^^^
NameError: name 'select_load_function' is not defined

ERROR: cli.azure.cli.core.azclierror: The command failed with an unexpected error. Here is the traceback:
ERROR: az_command_data_logger: The command failed with an unexpected error. Here is the traceback:
ERROR: cli.azure.cli.core.azclierror: name 'select_load_function' is not defined
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 664, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 731, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 701, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 334, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 240, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 453, in cached_put
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 447, in _put_operation
  File "C:\Users\mstetka\.azure\cliextensions\ml\azext_mlv2\manual\custom\workspace.py", line 212, in ml_workspace_update
    workspace = select_load_function()(file, params_override=params_override)
                ^^^^^^^^^^^^^^^^^^^^
NameError: name 'select_load_function' is not defined
ERROR: az_command_data_logger: name 'select_load_function' is not defined
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 664, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 731, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 701, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 334, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 240, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 453, in cached_put
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 447, in _put_operation
  File "C:\Users\mstetka\.azure\cliextensions\ml\azext_mlv2\manual\custom\workspace.py", line 212, in ml_workspace_update
    workspace = select_load_function()(file, params_override=params_override)
                ^^^^^^^^^^^^^^^^^^^^
NameError: name 'select_load_function' is not defined
To check existing issues, please visit: https://github.com/Azure/azure-cli/issues
DEBUG: cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x000002883BD9E5C0>]
INFO: az_command_data_logger: exit code: 1
INFO: cli.__main__: Command ran in 3.356 seconds (init: 0.358, invoke: 2.999)
INFO: telemetry.main: Begin splitting cli events and extra events, total events: 1
INFO: telemetry.client: Accumulated 0 events. Flush the clients.
INFO: telemetry.main: Finish splitting cli events and extra events, cli events: 1
INFO: telemetry.save: Save telemetry record of length 7676 in cache
INFO: telemetry.main: Begin creating telemetry upload process.
INFO: telemetry.process: Creating upload process: "C:\Program Files\Microsoft SDKs\Azure\CLI2\python.exe C:\Program Files\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\telemetry\__init__.pyc C:\Users\mstetka\.azure"
INFO: telemetry.process: Return from creating process
INFO: telemetry.main: Finish creating telemetry upload process.

### Expected behavior

For this particular command: An error stating that no configuration changes have been provided.
When issued with the --file argument that has a valid YAML configuration file, the workspace should be updated with the configuration

### Environment Summary

azure-cli                         2.61.0

core                              2.61.0
telemetry                          1.1.0

Extensions:
ml                                2.26.1

Dependencies:
msal                              1.28.0
azure-mgmt-resource               23.1.1

Python location 'C:\Program Files\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\mstetka\.azure\cliextensions'

Python (Windows) 3.11.8 (tags/v3.11.8:db85d51, Feb  6 2024, 22:03:32) [MSC v.1937 64 bit (AMD64)]

Legal docs and information: aka.ms/AzureCliLegal

Your CLI is up-to-date.

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by diondrapeck** [comment]
@FrsECM could you try again? This seems related to[ this issue](https://github.com/Azure/azure-cli/issues/28980).

**Comment by mstetka-fr** [comment]
@diondrapeck I do not see how this issue is related to that one, except it apparently was introduced in the 2.26.0 version of the extension. This problem persists in 2.26.1, but is not present in 2.25.0

I was able to get the command to work locally by installing version 2.25.0

```
az extension remove --name ml
az extension add --name ml --version 2.25.0
```

**Comment by diondrapeck** [comment]
Yes, @mstetka-fr. We ultimately determined this was an issue in the intermediate release and is now resolved in 2.25.0.

**Comment by diondrapeck** [comment]
This will be fixed in the 2.27.0 release for ml cli.
